auth_keycloak/wizard/keycloak_sync_wiz.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from odoo import fields, models, api, exceptions, _
+import logging
+import requests
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:
+    # py2
+    JSONDecodeError = ValueError
+
+logger = logging.getLogger(__name__)
+
+
+class KeycloakSyncMixin(models.TransientModel):
+    """Synchronize Keycloak users mixin."""
+
+    _name = 'auth.keycloak.sync.mixin'
+
+    provider_id = fields.Many2one(
+        string='Provider',
+        comodel_name='auth.oauth.provider',
+        required=True,
+    )
+    management_enabled = fields.Boolean(
+        related='provider_id.users_management_enabled',
+        readonly=True,
+    )
+    endpoint = fields.Char(
+        related='provider_id.users_endpoint',
+        readonly=True,
+    )
+    user = fields.Char(
+        related='provider_id.superuser',
+        readonly=True,
+    )
+    pwd = fields.Char(
+        related='provider_id.superuser_pwd',
+        readonly=True,
+    )
+    # tech field to map unique key from Keycloak to Odoo
+    login_match_key = fields.Selection(
+        selection=[
+            # keycloak:odoo
+            ('username:login', 'username'),
+            ('email:partner_id.email', 'email'),
```

Comment:
As it's a mixin shouldn't you define login and partner_id fields here ?

Reply:
uh? this is a mapping to match internal fields w/ external fields. What do you mean?

Reply:
Actually considering https://github.com/OCA/server-auth/pull/48/files#diff-3ca04c3a0916ac2ee82f250785800931R110 I should have asked if you shouldn't define `username` and `email` here.
But having another look at it, if it's only to be used here, it makes sense.